specification.md

```diff
@@ -1059,6 +1061,140 @@ Note that the referenced function definition type in this case must be `rest` (d
+
+The table below lists how to describe the HTTP parameters in the `operation` attribute:
+
+| Parameter | Description | Example |
+| --- | --- | --- |
+| Path | Path parameters are part of the location and must be enclosed with brackets (`{}`). The parameter name can be later referenced by the workflow states. | `/user/{id}` |
+| Query | [Query parameters](https://www.rfc-editor.org/rfc/rfc9110.html#section-4.2.2) must be enclosed with brackets (`{}`) after the query key separator (`=`). The parameter name can be later referenced by the workflow states. | `/user?status={status}` |
+
+Additional HTTP Headers can be passed via the `metadata` function definition. For example:
+
+```json
+{
+"functions": [
+  {
+    "name": "queryUserById",
+    "operation": "GET#/users/{id}",
+    "type": "http",
+    "metadata": {
+      "x-custom-header": "the-value"
+    }
+  }
+]
+}
+```
+
+> Runtimes are encouraged to support at least `Content-type: application/json` header when making HTTP invocations.
+
+Avoid passing sensible information in headers such as authentication tokens. See the [auth definition](#auth-definition) for more information.
+
+HTTP request body doesn't need to be described in the function definition. **All the data within the state** should be passed to the HTTP request. See the [action data filter](#action-data-filters) if you need to limit the amount of data passed to a given action.
+
+The function can be referenced during workflow execution when the invocation of the HTTP service is desired. For example:
+
+```json
+{
+"states": [
+  {
+      "name":"QueryUserInfo",
+      "type":"operation",
+      "actions":[
+       {
+          "functionRef": "queryUserById",
+          "arguments": {
+            "id": "${ .user.id }"
+          }
+      }],
+      "end": true
+  }]
+}
+```
+
+Example of `POST` request sending the state data as part of the body:
+
+```json
+{
+"functions": [
+  {
+    "name": "createUser",
+    "operation": "POST#/users",
+    "type": "http"
+  }
+]
+}
+```
+
+Then you can reference the `createUser` function and filter the data to invoke it.
+
+Data example:
+
+```json
+{
+  "order": { "id": "1234N", "products": [{"name": "Product 1"}]},
+  "user": { "name": "John Doe", "email": "john@doe.com" }
+}
+```
+
+Function invocation example:
+
+```json
+{
+"states": [
+  {
+      "name":"CreateNewUser",
+      "type":"operation",
+      "actions":[
+       {
+       "functionRef": "createUser",
+       "actionDataFilter": {
+          "fromStateData": "${ .user }",
+          "toStateData": "${ .user.id }"
+       }
+      }],
+      "end": true
+  }]
+}
+```
+
+In this case, only the contents of the `user` attribute will be passed to the function. The user ID returned by the HTTP request body will then be added to the state data:
```

Comment:
> only the contents of the `user` attribute will be passed to the function

What controls the format?

Reply:
Do you mean the content format? I don't follow. Everything is JSON from the spec perspective.

Reply:
That is non-obvious when talking about generic HTTP requests, and might well need to change in the future.